File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string ans = "";
        int sz = num.length();
        
        for(int i=0; i<sz-2; i++)
            if(num[i] == num[i+1] && num[i] == num[i+2]) ans = max(ans, num.substr(i, 3));
        
        return ans;
    }
};

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        int sz = num.length();
        String ans = "";
        StringBuilder str = new StringBuilder("");

        for(int i=0; i<sz; i++){
            str.append(num.charAt(i));
            if(i >= 2){
                if(str.charAt(0) == str.charAt(1) && str.charAt(1) == str.charAt(2)){
                    ans = ans.compareTo(str.toString()) >= 0 ? ans : str.toString();
                }
                str.deleteCharAt(0);
            }
        }

        return ans.toString();
    }
}